app/lib/features/home/providers/navigation.dart

```diff
@@ -0,0 +1,229 @@
+            label: Text(
+              roomId,
+              style: Theme.of(context).textTheme.labelSmall,
+              softWrap: false,
+            ),
+            location: '/$roomId',
+          ),
+          error: (err, _trace) => SidebarNavigationItem(
+            icon: const Icon(Atlas.warning_bold),
+            label: Text(
+              '$roomId: $err',
+              style: Theme.of(context).textTheme.labelSmall,
+              softWrap: false,
+            ),
+            location: '/$roomId',
+          ),
+          data: (info) => SidebarNavigationItem(
+            icon: ActerAvatar(
+              uniqueId: roomId,
+              displayName: info.displayName,
+              mode: DisplayMode.Space,
+              avatar: info.getAvatarImage(),
+              size: 24,
+            ),
+            label: Text(
+              info.displayName,
+              style: Theme.of(context).textTheme.labelSmall,
+              softWrap: false,
+            ),
+            location: '/$roomId',
+          ),
+        );
+      }).toList();
+    },
+  );
+});
+
+// provider that returns a string value
+final sidebarItemsProvider =
+    Provider.family<List<SidebarNavigationItem>, BuildContext>((ref, context) {
+  AsyncValue<List<SidebarNavigationItem>> config =
+      ref.watch(spaceItemsProvider(context));
+
+  final features = [
+    SidebarNavigationItem(
+      icon: const Icon(Atlas.magnifying_glass_thin),
+      label: Text(
+        'Search',
+        style: Theme.of(context).textTheme.labelSmall,
+        softWrap: false,
+      ),
+      location: Routes.quickJump.route,
+      pushToNavigate: true,
+    ),
+    SidebarNavigationItem(
+      icon: const Icon(Atlas.home_thin),
+      label: Text(
+        'Overview',
+        style: Theme.of(context).textTheme.labelSmall,
+        softWrap: false,
+      ),
+      location: Routes.dashboard.route,
+    ),
+    SidebarNavigationItem(
+      // icon: const Badge(child: Icon(Atlas.chats_thin)), // TODO: Badge example
+      icon: const Icon(Atlas.chats_thin),
+      label: Text(
+        'Chat',
+        style: Theme.of(context).textTheme.labelSmall,
+        softWrap: false,
+      ),
+      location: Routes.chat.route,
+    ),
+    SidebarNavigationItem(
+      icon: const Icon(Atlas.audio_wave_thin),
+      label: Column(
+        children: [
+          Text(
+            'Activities',
+            style: Theme.of(context).textTheme.labelSmall,
+            softWrap: false,
+          ),
+          const SizedBox(height: 10),
+          const Divider(
+            indent: 10,
+            endIndent: 10,
+          )
+        ],
+      ),
+      location: Routes.activities.route,
+    ),
+  ];
+
+  return config.when(
+    loading: () => features,
+    error: (err, stack) => features,
+    data: (spaces) {
+      if (spaces.isEmpty) {
+        return features;
+      }
+      return [...features, ...spaces];
+    },
+  );
+});
+
+final currentSelectedSidebarIndexProvider =
```

Comment:
A lot cleaner way in my mind would go something like having ```BottomBarNavigationItem``` as immutable state object (achievable) with freezed and doing interaction calls inside ```StateNotifierProvider<BottomBarNavigationItem>```.

Reply:
the currentlocation notifications was quite a tricky part and I am not sure I get - entirely - what you are proposing but I am open to trying it.

Reply:
See the ```bug_report_state.dart``` where we have ```BugReport``` freezed class and StateNotifier controlling state object.

Reply:
I see, but the problem is that we don't have an interaction to update the index but instead are basing that off `ref.watch(currentLocationProvider)` ....

not sure how you want to use a freezed class here or what benefit that has ...?

Reply:
Oh...I was thinking maybe location updates based on user navigation.